resources/assets/js/markdown-editor/plugins/youtube.js

```diff
@@ -0,0 +1,221 @@
+
+    const popupContent = createPopupContent(editor);
+
+    createPopup(editor, name, menuIndex, svgIcon, popupContent, title, tooltip);
+};
+
+const convertHtmlToMarkdown = (html) => {
+    const regex = new RegExp(`<img\\s+[^>]*src="youtube:([^"]*)"[^>]*>`, "gm");
+
+    // Used to ensure that the root element is a DIV so its produce valid HTML
+    // This avoid some strange behaviours on the preview
+    const wrapInDivRegex = new RegExp(
+        `<([A-Z][A-Z0-9]*)[^>]*data-nodeid="([^"]*)"[^>]*>(((?!<\\1)[\\s\\S])*<img\\s+[^>]*src="youtube:[^"]*"[^>]*>.*?(<\/\\1|))<\/\\1>`,
+        "gmsi"
+    );
+    const validHTML = html.replace(
+        wrapInDivRegex,
+        '<div data-nodeid="$2">$3</div>'
+    );
+
+    let replacemenent = validHTML;
+    let matches;
+    while ((matches = regex.exec(validHTML)) !== null) {
+        // This is necessary to avoid infinite loops with zero-width matches
+        if (matches.index === regex.lastIndex) {
+            regex.lastIndex++;
+        }
+
+        if (matches.length && matches.length >= 1) {
+            const youtubeCode = matches[1];
+            const regexToReplace = new RegExp(
+                `<img\\s+[^>]*src="youtube:(${escapeRegExp(
+                    youtubeCode
+                )})"[^>]*>`,
+                "gm"
+            );
+            const embed = getYoutubeEmbedCode(youtubeCode);
+
+            replacemenent = replacemenent.replace(regexToReplace, embed);
+        }
+    }
+
+    return replacemenent;
+};
+
+const convertMarkdownToHtml = (markdown) => {
+    const regex = new RegExp(
+        `<iframe[^>]*src\\s*=\\s*"?https?:\/\/[^\\s"\/]*youtube.com\/embed\/((?:[^\\s"]*)?)"?[^>]*>.*?<\/iframe>`,
+        "gm"
+    );
+    let matches;
+
+    let replacemenent = markdown;
+    while ((matches = regex.exec(markdown)) !== null) {
+        // This is necessary to avoid infinite loops with zero-width matches
+        if (matches.index === regex.lastIndex) {
+            regex.lastIndex++;
+        }
+
+        if (matches.length && matches.length >= 1) {
+            const youtubeCode = matches[1];
+            const regexToReplace = new RegExp(
+                `<iframe[^>]*src\\s*=\\s*"?https?:\/\/[^\\s"\/]*youtube.com\/embed\/(${escapeRegExp(
+                    youtubeCode
+                )})"?[^>]*>.*?<\/iframe>`,
+                "gm"
+            );
+            const code = getYoutubeMarkdown(youtubeCode);
+
+            replacemenent = replacemenent.replace(regexToReplace, code);
+        }
+    }
+
+    return replacemenent;
+};
+
+const addYoutubeMarkdownCommand = (mde, code) => {
+    if (!code) {
+        return;
+    }
+
+    const cm = mde.getEditor();
+
+    const rangeFrom = cm.getCursor("from");
+    const rangeTo = cm.getCursor("to");
+    const text = getYoutubeMarkdown(code);
+
+    cm.replaceSelection(text, "start", 0);
+
+    cm.setSelection(
+        {
+            line: rangeFrom.line,
+            ch: rangeFrom.ch + text.length,
```

Comment:
what's `ch`? can it be more descriptive or does the editor framework expect this property?